controllers/scraper.go

```diff
@@ -0,0 +1,60 @@
+package controllers
+
+import (
+	"net/http"
+
+	form "go-crawler-challenge/forms/scrapper"
+	"go-crawler-challenge/services/scraper"
+
+	"github.com/beego/beego/v2/server/web"
+)
+
+// ScraperController operations for Dashboard
+type ScraperController struct {
+	BaseController
+}
+
+// NestPrepare prepares some configurations to the controller
+func (c *ScraperController) NestPrepare() {
+	c.actionPolicyMapping()
+}
+
+// URLMapping maps search controller actions to functions
+func (c *ScraperController) URLMapping() {
+	c.Mapping("Create", c.Create)
+}
+
+// actionPolicyMapping maps search controller actions to policies
+func (c *ScraperController) actionPolicyMapping() {
+	c.MappingPolicy("Create", Policy{requireAuthenticatedUser: true})
```

Comment:
I think we can make it dynamic like this pseudo code 

```
c.MappingPolicy(actionName, Policy{canAccess: isAuthenticatedUser()})
```
So that we dont need to write it manually for every method like you did in the [session controller](https://github.com/Lahphim/go-crawler-challenge/blob/development/controllers/session.go#L30-L34)
```go
c.MappingPolicy("New", Policy{requireGuestUser: true})
c.MappingPolicy("Create", Policy{requireGuestUser: true})
c.MappingPolicy("Delete", Policy{requireAuthenticatedUser: true})
```

as `NestPrepare()` is executing every request, so we just need to set policy for the current request(current requested controller action)

Reply:
@junan I do like this because I want to keep consistency with the codebase I have around here https://github.com/Lahphim/go-crawler-challenge/pull/54/files#diff-38ccd6ad4383b82c9ef8a430ec6e8bc774fc40fa56893d01cfe3171e45d8b6a5R23

I think the current one looks clean for me 😇